File: ios/StoreReviewModule.m
//
//  StoreReviewModule.m
//  ReactNativeStoreReview
//
//  Created by Nayara Dias Gonçalves on 06/07/23.
//

#import <Foundation/Foundation.h>
#import <StoreKit/SKStoreReviewController.h>
#import <UIKit/UIKit.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(StoreReviewModule, NSObject)

- (dispatch_queue_t)methodQueue
{
  return dispatch_get_main_queue();
}

RCT_EXTERN_METHOD(requestReview)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

@end

File: ios/StoreReviewModule.swift
//
//  StoreReviewModule.swift
//  ReactNativeStoreReview
//
//  Created by Nayara Dias Gonçalves on 06/07/23.
//

import Foundation
import UIKit
import StoreKit

@objc(StoreReviewModule)
class StoreReviewModule: NSObject {

  @objc
  func requestReview () -> Void {

    if #available(iOS 10.3, *) {
        SKStoreReviewController.requestReview()
    } 
  }

  @objc
  func constantsToExport() -> [String: Any]! {
    return ["isAvailable":  (NSClassFromString("SKStoreReviewController") != nil)  ? (true) : (false)]
  }
}
